### DISCUSSION TITLE
请求提取BT 主题“1.2背景设置”片段

### DISCUSSION THREAD

**Discussion by lukyluki** [ProudBenzene/Blue-Topaz-Legacy#13] [Ideas]
大佬，能否麻烦你提取一下BT 主题中“1.2背景设置”的CSS片段？希望能在Minimal主题中使用笔记背景动态波浪的效果

<img width="582" alt="%7BCC1A3959-755A-48ed-A8CB-2453BFECEBB3%7D" src="https://github.com/ProudBenzene/Blue-Topaz-Legacy/assets/81464943/15d59ffe-cd0e-4a55-99d8-b7673487e60c">

**Comment by ProudBenzene** [comment]
您好，这个效果我也觉得很漂亮。但是我个人最近不是很有空，而且这个设置项从功能角度剖析，仅仅只是单纯的美化，其他方面的功能性较少。我制作legacy这个片线的初衷主要是为了集成一些topaz主题中功能性较强的设置，便于在其他主题中调用。

这个效果我也很喜欢，如果之后有空可能可以尝试进行提取，最近实在有点忙，在此抱歉。我也建议您在社区中进行求助，可能社区中就有其他大佬对此同样很感兴趣。

**Comment by lukyluki** [comment]
感谢大佬的回复！这设置项确实仅是单纯的美化，就是最近我发现了一些效果不错的动态波浪效果；想通过仿照BT 主题中“1.2背景设置”里的设置效果，把新的效果加进去后编写一个通用的CSS片段应用到更多的主题中

**Comment by ProudBenzene** [comment]
好的，如果顺利，欢迎您进行PR

**Comment by lukyluki** [comment]
大佬 我试着提取了一下BT [主题中“1.2背景设置”里的动态波浪效果，也在Q群里找人帮了些忙；提取的代码如下：

**Comment by lukyluki** [comment]
/*蓝色托帕石，适合Obsidian v1.2.8。uid [20230802] flying fly flies制作。如果喜欢，可以在Github里给我星星 https://github.com/whyt-byte */
/*Blue Topaz for Obsidian v1.2.8, uid [20230802] created by flying fly flies. Feel free to use, share and modify. If you like the theme, please star me on GitHub https://github.com/whyt-byte */

/* 致谢 Credits：
thanks pkmer.cn
Many thanks to @Cuman
@LillianWho @成雙酱 @嘴上云 @Klaas @Thinkbond @Rainbell129(AKA Lavi) @awyugan @GeoffreyOuO (WooYoo~~~) @锋华 @CZ-15/YZ-10 */

/* @settings
name: 🥑Blue Topaz Theme
id: blue-topaz-theme
settings:
  -
    id: background-settings
    title: 1. General Settings
    title.zh: 1. 整体性设置
    type: heading
    level: 1
    collapsed: false
  -
    id: color-settings
    title: 1.1 Color scheme
    title.zh: 1.1 整体配色
    type: heading
    level: 2
    collapsed: true
  -
    id: color-scheme-options
    title: Color Palette (from Topaz Community)
    title.zh: 整体配色选择 (由Topaz社区贡献)
    type: class-select
    allowEmpty: false
    default: color-scheme-options-default
    options:
      -
        label: Default
        value: color-scheme-options-default
      -
        label: Avocado
        value: color-scheme-options-avocado-topaz
      -
        label: Monochrome
        value: color-scheme-options-monochrome-topaz
      -
        label: Pink
        value: color-scheme-options-pink-topaz
      -
        label: Topaz-Nord
        value: color-scheme-options-topaz-nord
      -
        label: =↓😺 Created by Topaz Community 🐵↓=
        value:
      -
        label: Flamingo (@Mouth on Cloud & @Rainbell)
        value: color-scheme-options-flamingo
      -
        label: Honey milk (@LillianWho)
        value: color-scheme-options-honey-milk-topaz
      -
        label: 榛子巧克力 Hazelnut chocolate (@LillianWho)
        value: color-scheme-options-chocolate-topaz
      -
        label: Lilac (@awyugan)
        value: color-scheme-options-lilac
      -
        label: Autumn (@LillianWho)
        value: color-scheme-options-autumn-topaz
      -
        label: 魔方 Rubik's Cube (@LillianWho)
        value: color-scheme-options-lillimon-topaz
      -
        label: Simplicity (@Cuman)
        value: color-scheme-options-simplicity-topaz
  -
    id: disable-auto-hide-header
    title:  Disable auto hide header
    title.zh: 禁用自动隐藏头部
    type: class-toggle
    default: false
  -
    id: color-magic
    title: Color of Rubik's Cube palette
    title.zh: 魔方配色
    description.zh: 当整体配色为【魔方】时生效
    type: heading
    level: 3
    collapsed: true
  -
    id: magic-main-color
    title: main color
    title.zh: 主题色
    description.zh: 主题色主要作用于标题，字体，列表等主色调，建议用对比度强，不易疲劳的色系。如果想单独调整个别类目请在2.1中继续细调。
    type: variable-themed-color
    opacity: true
    format: hex
    default-light: '#'
    default-dark: '#'
  -
    id: magic-low-color
    title: Low contrast color
    title.zh: 浅色
    description.zh: 浅色主要作用于各类触发背景底色，建议用不影响阅读的背景色系。
    type: variable-themed-color
    opacity: true
    format: hex
    default-light: '#'
    default-dark: '#'
  -
    id: magic-deep-color
    title: deep color
    title.zh: 深色
    description.zh: 深色主要作用于主题强调色系，用在各种控件，提示等使用，建议用浅色相关的醒目色系。
    type: variable-themed-color
    opacity: true
    format: hex
    default-light: '#'
    default-dark: '#'
  -
    id: background-settings
    title: 1.2 Background Images
    title.zh: 1.2 背景设置
    type: heading
    level: 2
    collapsed: true
  -
    id: background-settings-workplace
    title: 1.2.1 Workplace Background Image
    title.zh: 1.2.1 工作页面背景
    type: heading
    level: 3
    collapsed: true
  -
    id: background-settings-workplace-background-image
    title:  Activate Image Background
    title.zh: 开启背景
    type: class-toggle
    default: false
    addCommand: true
  -
    id: background-settings-workplace-theme-light
    title: Light mode
    type: class-select
    allowEmpty: false
    default: background-settings-workplace-theme-light-leaf
    options:
      -
        label: Blue Mountain
        value: background-settings-workplace-theme-light-blue-mountain
      -
        label: In the sky
        value: background-settings-workplace-theme-light-in-the-sky
      -
        label: Leaf
        value: background-settings-workplace-theme-light-leaf
      -
        label: Random
        value: background-settings-workplace-random-light
      -
        label: Waves
        value: background-settings-workplace-waves-light
      -
        label: Animating waves
        value: background-settings-workplace-waves2-light
      -
        label: Custom
        value: background-settings-workplace-theme-light-custom-option
  
*/
body {

  --simple-tran:hsla(204, 45%, 97%, 0);
  --blur-depth: 10px;
  --blur-depth-cp: 10px;
  --saturate-4-bg: 1;
  --saturate-4-bg-cp: 1;
  --brightness-4-bg: 0.9;
  --brightness-4-bg-cp: 0.9;
  --opacity-cp: 0.25;
  --titlebar-brightness: 1;
  --titlebar-blur-depth: 8px;
  --stack-tabs-background-blur-depth:40px;
  --background-waves-color:#4e7be58c;
  --simple-white-2:hsla(210, 45%, 95%, 1);
  --background-secondary-bg-4-bt: var(--simple-tran);
  --background-primary: var(--background-primary-bg-4-bt,#ffffff);
  --background-primary-alt: var(--background-primary-alt-bg-4-bt,#e9e9e9);
  --background-secondary: var(--background-secondary-bg-4-bt,#fcfcfc);
  --background-secondary-alt: var(--background-secondary-alt-bg-4-bt,#f3f3f3);
  --background-settings-workplace-theme-light-custom: url("https://");
  --background-settings-workplace-theme-dark-custom: url("https://");
}

body.color-scheme-options-simplicity-topaz.background-settings-workplace-background-image:is(.background-settings-workplace-waves2-light,.background-settings-workplace-waves-light).theme-light{
  --background-secondary-bg-4-bt: var(--simple-tran);
}

body:not(.simple-titlebar).background-settings-workplace-waves-light.theme-light .titlebar::before {
  content: '';
  display: block;
  position: absolute;
  height:40px;
  width: 120%;
  background-image: unset;
  -webkit-mask-image: var(--theme-background);
  background-size: 100% 100%;
  opacity: 0.5;
  background-color: var(--wave-background-color);
}

body.background-settings-workplace-background-image:not(.simple-titlebar):not(.is-frameless) .titlebar::before {
  background-image: var(--theme-background);
  background-size: cover;
  content: '';
  display: block;
  position: absolute;
  top: -3px;
  filter: blur(var(--titlebar-blur-depth)) brightness(var(--titlebar-brightness)) opacity(0.5);
  right: -15px;
  bottom: 0px;
  left: -15px;
}
body.background-settings-workplace-background-image:not(.simple-titlebar):not(.is-frameless).theme-dark .titlebar::before {
  filter: blur(var(--titlebar-blur-depth)) brightness(calc(var(--titlebar-brightness) - 0.5)) opacity(0.5);
}

body.is-mobile.background-settings-workplace-background-image:not(:is(.background-settings-workplace-waves-light,.background-settings-workplace-waves2-light,.background-settings-workplace-waves,.background-settings-workplace-waves2)) {
  background-image: var(--theme-background);
  background-size: cover;
  opacity: 0.8;
}

body:not(.simple-titlebar).background-settings-workplace-waves-light.theme-light .titlebar::before {
  content: '';
  display: block;
  position: absolute;
  height:40px;
  width: 120%;
  background-image: unset;
  -webkit-mask-image: var(--theme-background);
  background-size: 100% 100%;
  opacity: 0.5;
  background-color: var(--wave-background-color);
}

body.is-mobile.background-settings-workplace-background-image:is(.background-settings-workplace-waves-light,.background-settings-workplace-waves2-light,.background-settings-workplace-waves,.background-settings-workplace-waves2),
body.background-settings-workplace-background-image:is(.background-settings-workplace-waves-light,.background-settings-workplace-waves2-light).theme-light .workspace,
body.color-scheme-options-simplicity-topaz.background-settings-workplace-background-image:is(.background-settings-workplace-waves,.background-settings-workplace-waves2).theme-dark .workspace,
body.background-settings-workplace-background-image:is(.background-settings-workplace-waves-light,.background-settings-workplace-waves2-light):not(.is-hidden-frameless).theme-light .titlebar {
  background: var(--simple-white-2);
}

body.background-settings-workplace-background-image:is(.background-settings-workplace-waves-light,.background-settings-workplace-waves,.background-settings-workplace-waves2,.background-settings-workplace-waves2-light):not(.is-hidden-frameless) .titlebar-inner,
body.background-settings-workplace-background-image:is(.background-settings-workplace-waves,.background-settings-workplace-waves2).theme-dark .workspace {
  background: var(--background-secondary);
}

/****from https://www.shapedivider.app/*****/
body.background-settings-workplace-background-image.background-settings-workplace-waves-light.theme-light .workspace:before,
body.background-settings-workplace-background-image.background-settings-workplace-waves.theme-dark .workspace:before {
  content: "";
  position: fixed;
  height: auto;
  width: 300%;
  transform: rotate(180deg);
  top: 75%;
  left: -25%;
  background-image: unset;
  -webkit-mask-image: var(--theme-background);
  background-size: 100% 100%;
  opacity: 0.7;
  filter: blur(var(--blur-depth)) brightness(var(--brightness-4-bg)) saturate(var(--saturate-4-bg));
  background-color: var(--wave-background-color);
}
/**from https://svgwave.in/*/
body.background-settings-workplace-background-image.background-settings-workplace-waves2-light.theme-light .workspace::before,
body.background-settings-workplace-background-image.background-settings-workplace-waves2.theme-dark .workspace:before {
  content: "";
  position: fixed;
  height: auto;
  width: 300%;
  top: 75%;
  left: -25%;
  background-image: unset;
  -webkit-mask-image: var(--theme-background);
  background-size: 100% 100%;
  opacity: 0.7;
  filter: blur(var(--blur-depth)) brightness(var(--brightness-4-bg)) saturate(var(--saturate-4-bg));
  background-color: var(--wave2-background-color);
}

body.background-settings-workplace-background-image.theme-light .workspace-split.mod-root:fullscreen::before,
body.background-settings-workplace-background-image.theme-dark .workspace-split.mod-root:fullscreen::before {
  background-image: var(--theme-background);
  background-size: cover;
  filter: blur(var(--blur-depth)) brightness(var(--brightness-4-bg)) saturate(var(--saturate-4-bg));
  content: '';
  display: block;
  position: absolute;
  top: -25px;
  right: -15px;
  bottom: 0px;
  left: -15px;
}

/****from https://www.shapedivider.app/*****/
body.background-settings-workplace-background-image.background-settings-workplace-waves-light.theme-light .workspace-split.mod-root::before,
body.background-settings-workplace-background-image.background-settings-workplace-waves.theme-dark .workspace-split.mod-root::before {
  content: "";
  position: fixed;
  height: auto;
  width: 300%;
  transform: rotate(180deg);
  top: 75%;
  left: -25%;
  background-image: unset;
  -webkit-mask-image: var(--theme-background);
  background-size: 100% 100%;
  opacity: 0.7;
  filter: blur(var(--blur-depth)) brightness(var(--brightness-4-bg)) saturate(var(--saturate-4-bg));
  background-color: var(--wave-background-color);
}

/**from https://svgwave.in/*/
body.background-settings-workplace-background-image.background-settings-workplace-waves2-light.theme-light .workspace-split.mod-root::before,
body.background-settings-workplace-background-image.background-settings-workplace-waves2.theme-dark .workspace-split.mod-root:before {
  content: "";
  position: fixed;
  height: auto;
  width: 300%;
  top: 75%;
  left: -25%;
  background-image: unset;
  -webkit-mask-image: var(--theme-background);
  background-size: 100% 100%;
  opacity: 0.7;
  filter: blur(var(--blur-depth)) brightness(var(--brightness-4-bg)) saturate(var(--saturate-4-bg));
  background-color: var(--wave2-background-color);
}

.workspace-split.mod-root:fullscreen .modal-container .prompt {
  width: 90%;
}

body.background-settings-workplace-background-image .workspace::before {
  content: '';
  display: block;
  position: absolute;
  top: -37px;
  right: -15px;
  bottom: 0;
  left: -15px;
}


body.background-settings-workplace-background-image.background-settings-workplace-waves-light.theme-light,
body.background-settings-workplace-background-image.background-settings-workplace-waves.theme-dark {
  --wave-background: url("data:image/svg+xml, %3Csvg data-name='Layer 1' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1200 120' preserveAspectRatio='none'%3E %3Cpath d='M0,0V46.29c47.79,22.2,103.59,32.17,158,28,70.36-5.37,136.33-33.31,206.8-37.5C438.64,32.43,512.34,53.67,583,72.05c69.27,18,138.3,24.88,209.4,13.08,36.15-6,69.85-17.84,104.45-29.34C989.49,25,1113-14.29,1200,52.47V0Z' opacity='.25' class='shape-fill'%3E%3C/path%3E %3Cpath d='M0,0V15.81C13,36.92,27.64,56.86,47.69,72.05,99.41,111.27,165,111,224.58,91.58c31.15-10.15,60.09-26.07,89.67-39.8,40.92-19,84.73-46,130.83-49.67,36.26-2.85,70.9,9.42,98.6,31.56,31.77,25.39,62.32,62,103.63,73,40.44,10.79,81.35-6.69,119.13-24.28s75.16-39,116.92-43.05c59.73-5.85,113.28,22.88,168.9,38.84,30.2,8.66,59,6.17,87.09-7.5,22.43-10.89,48-26.93,60.65-49.24V0Z' opacity='.5' class='shape-fill'%3E%3C/path%3E %3Cpath d='M0,0V5.63C149.93,59,314.09,71.32,475.83,42.57c43-7.64,84.23-20.12,127.61-26.46,59-8.63,112.48,12.24,165.56,35.4C827.93,77.22,886,95.24,951.2,90c86.53-7,172.46-45.71,248.8-84.81V0Z' class='shape-fill'%3E%3C/path%3E %3C/svg%3E");
  --theme-background: var(--wave-background);
  --wave-background-color: var(--background-waves-color);
}

body.background-settings-workplace-background-image.background-settings-workplace-waves2-light.theme-light,
body.background-settings-workplace-background-image.background-settings-workplace-waves2.theme-dark {
  --wave2-background: url("data:image/svg+xml,%3Csvg width='100%25' height='100%25' id='svg' viewBox='0 0 1440 600' xmlns='http://www.w3.org/2000/svg' class='transition duration-300 ease-in-out delay-150'%3E%3Cstyle%3E .path-0%7B animation:pathAnim-0 4s; animation-timing-function: linear; animation-iteration-count: infinite; %7D @keyframes pathAnim-0%7B 0%25%7B d: path('M 0,600 C 0,600 0,150 0,150 C 81.35641025641027,172.9153846153846 162.71282051282054,195.83076923076922 252,194 C 341.28717948717946,192.16923076923078 438.50512820512824,165.59230769230766 511,146 C 583.4948717948718,126.40769230769234 631.2666666666665,113.80000000000004 703,116 C 774.7333333333335,118.19999999999996 870.428205128205,135.20769230769227 957,154 C 1043.571794871795,172.79230769230773 1121.0205128205127,193.3692307692308 1200,193 C 1278.9794871794873,192.6307692307692 1359.4897435897437,171.3153846153846 1440,150 C 1440,150 1440,600 1440,600 Z'); %7D 25%25%7B d: path('M 0,600 C 0,600 0,150 0,150 C 93.78717948717946,117.86410256410255 187.57435897435892,85.72820512820512 265,105 C 342.4256410256411,124.27179487179488 403.4897435897436,194.95128205128205 488,191 C 572.5102564102564,187.04871794871795 680.4666666666668,108.46666666666668 760,108 C 839.5333333333332,107.53333333333332 890.6435897435897,185.18205128205128 968,201 C 1045.3564102564103,216.81794871794872 1148.9589743589743,170.8051282051282 1232,152 C 1315.0410256410257,133.1948717948718 1377.520512820513,141.5974358974359 1440,150 C 1440,150 1440,600 1440,600 Z'); %7D 50%25%7B d: path('M 0,600 C 0,600 0,150 0,150 C 96.15641025641025,143.13589743589745 192.3128205128205,136.2717948717949 266,127 C 339.6871794871795,117.72820512820512 390.9051282051282,106.04871794871795 455,129 C 519.0948717948718,151.95128205128205 596.0666666666668,209.5333333333333 676,201 C 755.9333333333332,192.4666666666667 838.8282051282051,117.81794871794872 936,114 C 1033.1717948717949,110.18205128205128 1144.6205128205129,177.1948717948718 1231,195 C 1317.3794871794871,212.8051282051282 1378.6897435897436,181.4025641025641 1440,150 C 1440,150 1440,600 1440,600 Z'); %7D 75%25%7B d: path('M 0,600 C 0,600 0,150 0,150 C 55.16923076923078,140.31538461538463 110.33846153846156,130.63076923076923 198,135 C 285.66153846153844,139.36923076923077 405.8153846153846,157.7923076923077 496,167 C 586.1846153846154,176.2076923076923 646.4000000000001,176.2 733,158 C 819.5999999999999,139.8 932.5846153846155,103.40769230769229 1001,104 C 1069.4153846153845,104.59230769230771 1093.2615384615385,142.16923076923078 1159,156 C 1224.7384615384615,169.83076923076922 1332.3692307692309,159.9153846153846 1440,150 C 1440,150 1440,600 1440,600 Z'); %7D 100%25%7B d: path('M 0,600 C 0,600 0,150 0,150 C 81.35641025641027,172.9153846153846 162.71282051282054,195.83076923076922 252,194 C 341.28717948717946,192.16923076923078 438.50512820512824,165.59230769230766 511,146 C 583.4948717948718,126.40769230769234 631.2666666666665,113.80000000000004 703,116 C 774.7333333333335,118.19999999999996 870.428205128205,135.20769230769227 957,154 C 1043.571794871795,172.79230769230773 1121.0205128205127,193.3692307692308 1200,193 C 1278.9794871794873,192.6307692307692 1359.4897435897437,171.3153846153846 1440,150 C 1440,150 1440,600 1440,600 Z'); %7D %7D%3C/style%3E%3Cdefs%3E%3ClinearGradient id='gradient' x1='0%25' y1='50%25' x2='100%25' y2='50%25'%3E%3Cstop offset='5%25' stop-color='%23002bdc66'%3E%3C/stop%3E%3Cstop offset='95%25' stop-color='%2332ded466'%3E%3C/stop%3E%3C/linearGradient%3E%3C/defs%3E%3Cpath d='M 0,600 C 0,600 0,150 0,150 C 81.35641025641027,172.9153846153846 162.71282051282054,195.83076923076922 252,194 C 341.28717948717946,192.16923076923078 438.50512820512824,165.59230769230766 511,146 C 583.4948717948718,126.40769230769234 631.2666666666665,113.80000000000004 703,116 C 774.7333333333335,118.19999999999996 870.428205128205,135.20769230769227 957,154 C 1043.571794871795,172.79230769230773 1121.0205128205127,193.3692307692308 1200,193 C 1278.9794871794873,192.6307692307692 1359.4897435897437,171.3153846153846 1440,150 C 1440,150 1440,600 1440,600 Z' stroke='none' stroke-width='0' fill='url(%23gradient)' class='transition-all duration-300 ease-in-out delay-150 path-0'%3E%3C/path%3E%3Cstyle%3E .path-1%7B animation:pathAnim-1 4s; animation-timing-function: linear; animation-iteration-count: infinite; %7D @keyframes pathAnim-1%7B 0%25%7B d: path('M 0,600 C 0,600 0,300 0,300 C 82.94102564102565,295.425641025641 165.8820512820513,290.85128205128206 257,284 C 348.1179487179487,277.14871794871794 447.41282051282053,268.02051282051286 514,267 C 580.5871794871795,265.97948717948714 614.4666666666667,273.06666666666666 676,278 C 737.5333333333333,282.93333333333334 826.7205128205128,285.71282051282054 921,295 C 1015.2794871794872,304.28717948717946 1114.6512820512821,320.0820512820513 1202,322 C 1289.3487179487179,323.9179487179487 1364.674358974359,311.9589743589744 1440,300 C 1440,300 1440,600 1440,600 Z'); %7D 25%25%7B d: path('M 0,600 C 0,600 0,300 0,300 C 88.24102564102563,279.49487179487176 176.48205128205126,258.98974358974357 263,274 C 349.51794871794874,289.01025641025643 434.3128205128206,339.5358974358975 519,332 C 603.6871794871794,324.4641025641025 688.2666666666667,258.8666666666667 759,247 C 829.7333333333333,235.13333333333335 886.6205128205129,276.99743589743593 960,293 C 1033.3794871794871,309.00256410256407 1123.2512820512823,299.14358974358976 1206,296 C 1288.7487179487177,292.85641025641024 1364.374358974359,296.42820512820515 1440,300 C 1440,300 1440,600 1440,600 Z'); %7D 50%25%7B d: path('M 0,600 C 0,600 0,300 0,300 C 77.61025641025643,301.1948717948718 155.22051282051285,302.38974358974355 229,309 C 302.77948717948715,315.61025641025645 372.7282051282051,327.63589743589745 455,315 C 537.2717948717949,302.36410256410255 631.8666666666667,265.0666666666667 730,264 C 828.1333333333333,262.9333333333333 929.8051282051283,298.0974358974359 1003,313 C 1076.1948717948717,327.9025641025641 1120.9128205128204,322.54358974358973 1189,317 C 1257.0871794871796,311.45641025641027 1348.5435897435898,305.7282051282051 1440,300 C 1440,300 1440,600 1440,600 Z'); %7D 75%25%7B d: path('M 0,600 C 0,600 0,300 0,300 C 53.37692307692305,305.2846153846154 106.7538461538461,310.5692307692308 202,310 C 297.2461538461539,309.4307692307692 434.36153846153854,303.00769230769225 521,293 C 607.6384615384615,282.99230769230775 643.8,269.4 716,259 C 788.2,248.6 896.4384615384615,241.39230769230772 974,254 C 1051.5615384615385,266.6076923076923 1098.4461538461537,299.03076923076924 1171,310 C 1243.5538461538463,320.96923076923076 1341.7769230769231,310.4846153846154 1440,300 C 1440,300 1440,600 1440,600 Z'); %7D 100%25%7B d: path('M 0,600 C 0,600 0,300 0,300 C 82.94102564102565,295.425641025641 165.8820512820513,290.85128205128206 257,284 C 348.1179487179487,277.14871794871794 447.41282051282053,268.02051282051286 514,267 C 580.5871794871795,265.97948717948714 614.4666666666667,273.06666666666666 676,278 C 737.5333333333333,282.93333333333334 826.7205128205128,285.71282051282054 921,295 C 1015.2794871794872,304.28717948717946 1114.6512820512821,320.0820512820513 1202,322 C 1289.3487179487179,323.9179487179487 1364.674358974359,311.9589743589744 1440,300 C 1440,300 1440,600 1440,600 Z'); %7D %7D%3C/style%3E%3Cdefs%3E%3ClinearGradient id='gradient' x1='0%25' y1='50%25' x2='100%25' y2='50%25'%3E%3Cstop offset='5%25' stop-color='%23002bdc88'%3E%3C/stop%3E%3Cstop offset='95%25' stop-color='%2332ded488'%3E%3C/stop%3E%3C/linearGradient%3E%3C/defs%3E%3Cpath d='M 0,600 C 0,600 0,300 0,300 C 82.94102564102565,295.425641025641 165.8820512820513,290.85128205128206 257,284 C 348.1179487179487,277.14871794871794 447.41282051282053,268.02051282051286 514,267 C 580.5871794871795,265.97948717948714 614.4666666666667,273.06666666666666 676,278 C 737.5333333333333,282.93333333333334 826.7205128205128,285.71282051282054 921,295 C 1015.2794871794872,304.28717948717946 1114.6512820512821,320.0820512820513 1202,322 C 1289.3487179487179,323.9179487179487 1364.674358974359,311.9589743589744 1440,300 C 1440,300 1440,600 1440,600 Z' stroke='none' stroke-width='0' fill='url(%23gradient)' class='transition-all duration-300 ease-in-out delay-150 path-1'%3E%3C/path%3E%3Cstyle%3E .path-2%7B animation:pathAnim-2 4s; animation-timing-function: linear; animation-iteration-count: infinite; %7D @keyframes pathAnim-2%7B 0%25%7B d: path('M 0,600 C 0,600 0,450 0,450 C 76.1974358974359,467.39230769230767 152.3948717948718,484.7846153846154 220,481 C 287.6051282051282,477.2153846153846 346.6179487179487,452.25384615384615 441,452 C 535.3820512820513,451.74615384615385 665.1333333333334,476.2 762,477 C 858.8666666666666,477.8 922.8487179487179,454.94615384615383 984,443 C 1045.1512820512821,431.05384615384617 1103.4717948717948,430.0153846153846 1179,433 C 1254.5282051282052,435.9846153846154 1347.2641025641026,442.9923076923077 1440,450 C 1440,450 1440,600 1440,600 Z'); %7D 25%25%7B d: path('M 0,600 C 0,600 0,450 0,450 C 86.80769230769232,454.0974358974359 173.61538461538464,458.1948717948718 262,469 C 350.38461538461536,479.8051282051282 440.3461538461538,497.3179487179487 508,488 C 575.6538461538462,478.6820512820513 621,442.53333333333336 697,418 C 773,393.46666666666664 879.6538461538462,380.5487179487179 972,401 C 1064.3461538461538,421.4512820512821 1142.3846153846155,475.2717948717949 1218,489 C 1293.6153846153845,502.7282051282051 1366.8076923076924,476.36410256410255 1440,450 C 1440,450 1440,600 1440,600 Z'); %7D 50%25%7B d: path('M 0,600 C 0,600 0,450 0,450 C 79.76153846153846,435.7282051282051 159.52307692307693,421.45641025641027 229,422 C 298.4769230769231,422.54358974358973 357.6692307692308,437.9025641025641 443,442 C 528.3307692307692,446.0974358974359 639.8,438.93333333333334 720,424 C 800.2,409.06666666666666 849.1307692307691,386.36410256410255 925,394 C 1000.8692307692309,401.63589743589745 1103.676923076923,439.6102564102564 1194,454 C 1284.323076923077,468.3897435897436 1362.1615384615384,459.1948717948718 1440,450 C 1440,450 1440,600 1440,600 Z'); %7D 75%25%7B d: path('M 0,600 C 0,600 0,450 0,450 C 72.5102564102564,462.1435897435897 145.0205128205128,474.28717948717946 236,478 C 326.9794871794872,481.71282051282054 436.42820512820515,476.9948717948718 503,459 C 569.5717948717949,441.0051282051282 593.2666666666667,409.7333333333333 677,408 C 760.7333333333333,406.2666666666667 904.5051282051284,434.07179487179496 998,437 C 1091.4948717948716,439.92820512820504 1134.7128205128206,417.9794871794872 1200,416 C 1265.2871794871794,414.0205128205128 1352.6435897435897,432.0102564102564 1440,450 C 1440,450 1440,600 1440,600 Z'); %7D 100%25%7B d: path('M 0,600 C 0,600 0,450 0,450 C 76.1974358974359,467.39230769230767 152.3948717948718,484.7846153846154 220,481 C 287.6051282051282,477.2153846153846 346.6179487179487,452.25384615384615 441,452 C 535.3820512820513,451.74615384615385 665.1333333333334,476.2 762,477 C 858.8666666666666,477.8 922.8487179487179,454.94615384615383 984,443 C 1045.1512820512821,431.05384615384617 1103.4717948717948,430.0153846153846 1179,433 C 1254.5282051282052,435.9846153846154 1347.2641025641026,442.9923076923077 1440,450 C 1440,450 1440,600 1440,600 Z'); %7D %7D%3C/style%3E%3Cdefs%3E%3ClinearGradient id='gradient' x1='0%25' y1='50%25' x2='100%25' y2='50%25'%3E%3Cstop offset='5%25' stop-color='%23002bdcff'%3E%3C/stop%3E%3Cstop offset='95%25' stop-color='%2332ded4ff'%3E%3C/stop%3E%3C/linearGradient%3E%3C/defs%3E%3Cpath d='M 0,600 C 0,600 0,450 0,450 C 76.1974358974359,467.39230769230767 152.3948717948718,484.7846153846154 220,481 C 287.6051282051282,477.2153846153846 346.6179487179487,452.25384615384615 441,452 C 535.3820512820513,451.74615384615385 665.1333333333334,476.2 762,477 C 858.8666666666666,477.8 922.8487179487179,454.94615384615383 984,443 C 1045.1512820512821,431.05384615384617 1103.4717948717948,430.0153846153846 1179,433 C 1254.5282051282052,435.9846153846154 1347.2641025641026,442.9923076923077 1440,450 C 1440,450 1440,600 1440,600 Z' stroke='none' stroke-width='0' fill='url(%23gradient)' class='transition-all duration-300 ease-in-out delay-150 path-2'%3E%3C/path%3E%3C/svg%3E");
  --theme-background: var(--wave2-background);
  --wave2-background-color: var(--background-waves-color);
}

body.is-mobile.background-settings-workplace-background-image:is(.background-settings-workplace-waves-light,.background-settings-workplace-waves2-light).theme-light,
body.is-mobile.background-settings-workplace-background-image:is(.background-settings-workplace-waves,.background-settings-workplace-waves2).theme-dark {
  --wave2-background-color: #001ae0;
  --wave-background-color: #245eff;
}

body.background-image-settings-command-palette-same-as-workplace.background-image-settings-command-palette-switch.background-settings-workplace-waves2-light .modal-container .prompt::before,
body.background-image-settings-command-palette-same-as-workplace.background-image-settings-command-palette-switch.background-settings-workplace-waves2 .modal-container .prompt::before {
  top: 60%;
}

body.theme-light.background-image-settings-command-palette-same-as-workplace.background-image-settings-command-palette-switch.background-settings-workplace-waves-light .modal-container .prompt::before,
body.background-image-settings-command-palette-same-as-workplace.background-image-settings-command-palette-switch.background-settings-workplace-background-image.background-settings-workplace-waves.theme-dark .prompt:before {
  transform: rotate(180deg);
  top: 75%;
  background-image:unset;
  -webkit-mask-image: var(--theme-background);
  background-size:100% 100%;
  opacity:0.5;
  background-color: var(--wave-background-color);
}

**Comment by lukyluki** [comment]
可能是提取得太粗糙了，在minimal里需要配合从Mado中提取的圆角边框代码才能实现效果，能给点建议吗？
